### DISCUSSION TITLE
Does Milvus support write serializability?

### DISCUSSION THREAD

**Discussion by wilbertharriman** [milvus-io/milvus#23826] [Q&A and General discussion] [answered]
Hello Milvus team,
I am considering Milvus for my application that may involve multiple clients sending update requests to Milvus at the same time.

For example,
Let's say I have a collection that looks like this
| entity_id | embedding |
|-----------|--------------|
|101| [1, 1, 1, ...] |
|102| [0, 0, 1, ...] |
```
writer 1 updates vector 101 and 102
writer 2 updates vector 101 and 102
```
The final result I would like to see is 
```
For vector 101 and 102 to both have the updates from writer 1
Alternatively, both have the updates from writer 2
```
In other words, my application requires write serializability.

Would Milvus be able to guarantee this?

**Comment by yhmo** [comment, marked as answer]
For the same client, write serializability is guaranteed.
But not guaranteed for multiple clients.
